### PR TITLE
Sc feature connect api

### PR DESCRIPTION
**Pull request by bastiampos**
Data connected with Messari API.

https://user-images.githubusercontent.com/82401272/153236534-e13f0c37-ea59-42b3-a329-664f20255377.mov

### CODE REVIEW

**Review by tomasdematos — APPROVED** [review]

File: redux/actions/currenciesActions.ts [hunk 1]
```diff
@@ -20,31 +20,36 @@ export const getUserCurrenciesFromAsync = () => async (dispatch: Dispatch<IGetCu
...
-    ))
-
-    return (symbol.toLowerCase() === valueSearched.toLowerCase() ||  name.toLowerCase() === valueSearched.toLowerCase()) && !isAdded
-  })
+export const addCurrencyBySymbol = (valueSearched: string) => async (dispatch: Dispatch<IGetCurrenciesAction>, getState: () => IState) => {
+  const {userCurrencyList} = getState().currencies
 
-  if (newCrypto) {
-    const updatedUserCurrencies: ICurrency[] = [...userCurrencyList, newCrypto]
+  const host = 'https://data.messari.io/api/v1/assets'
```
Comment (ale1313) [inline redux/actions/currenciesActions.ts]: [CRITICAL] This should be placed on an environment variable

File: redux/actions/currenciesActions.ts [hunk 2]
```diff
@@ -20,31 +20,36 @@ export const getUserCurrenciesFromAsync = () => async (dispatch: Dispatch<IGetCu
...
+
+      if (!isAdded) {
+        const updatedList = [...userCurrencyList, response.data.data]
+        dispatch({ type: ADD_NEW_CRYPTOCURRENCY, payload: updatedList }) 
+
+        try {
+          await AsyncStorage.setItem(AsyncKeys.userCurrencyList, JSON.stringify(updatedList))
+        } catch (e) {
+          console.log(e)
+        }
```
Comment (ale1313) [inline redux/actions/currenciesActions.ts]: [minor] This logic should be placed in the component, as it was before

File: types/currency.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
   symbol:  string;
-  price: number;
-  pctg: number,
-  src: string
+  market_data: MarketData;
+}
+
+export interface MarketData {
+  price_usd:                        number;
+  percent_change_usd_last_24_hours: number;
```
Comment (ale1313) [inline types/currency.ts]: [minor] Do not leave spaces

File: redux/actions/currenciesActions.ts [hunk 4]
```diff
@@ -20,31 +20,36 @@ export const getUserCurrenciesFromAsync = () => async (dispatch: Dispatch<IGetCu
...
+        symbol.toLowerCase() === valueSearched.toLowerCase() ||  name.toLowerCase() === valueSearched.toLowerCase()
+      ))
+
+      if (!isAdded) {
+        const updatedList = [...userCurrencyList, response.data.data]
+        dispatch({ type: ADD_NEW_CRYPTOCURRENCY, payload: updatedList }) 
+
+        try {
+          await AsyncStorage.setItem(AsyncKeys.userCurrencyList, JSON.stringify(updatedList))
+        } catch (e) {
```
Comment (ale1313) [inline redux/actions/currenciesActions.ts]: [question] Does the backend returns a certain error response? Generally it follows this structure: `response.error`

File: redux/reducers/currenciesReducer.ts [hunk 5]
```diff
@@ -2,50 +2,16 @@ import { IGetCurrenciesAction, ICurrenciesState } from "../../types"
...
   userCurrencyList: []
 }
 
-const currenciesReducer = ( state: ICurrenciesState = initialState, action: IGetCurrenciesAction): ICurrenciesState => {
-  switch (action.type) {
-    case GET_USER_CURRENCIES_FROM_ASYNC: 
+const currenciesReducer = ( state: ICurrenciesState = initialState, {type, payload}: IGetCurrenciesAction): ICurrenciesState => {
+  switch (type) {
+    case ADD_NEW_CRYPTOCURRENCY: 
+    case GET_USER_CURRENCIES_FROM_ASYNC:
```
Comment (ale1313) [inline redux/reducers/currenciesReducer.ts]: [minor] Those two cases shouldn't be one next to the other

File: components/CurrencyCard/index.tsx [hunk 6]
```diff
@@ -12,20 +12,26 @@ interface Props {
 const CurrencyCard: FC<Props> = ({currency}) => {
   const [isPositive, setIsPositive] = useState(false)
   const [price, setPrice] = useState('')
+  const [percentage, setPercentage] = useState(0)
+  const [srcPicture, setSrcPicture] = useState('`https://messari.io/asset-images/id/128.png?v=2`')
```
Comment (ale1313) [inline components/CurrencyCard/index.tsx]: [CRITICAL] This host should also be stored on a environment variable

File: components/CurrencyCard/index.tsx [hunk 7]
```diff
@@ -12,20 +12,26 @@ interface Props {
...
+  const {id, name, symbol, market_data: {price_usd, percent_change_usd_last_24_hours}} = currency
 
   useEffect(() => {
     const priceFormat = new Intl.NumberFormat('en-US')
-    setIsPositive(currency.pctg >= 0)
-    setPrice(priceFormat.format(currency.price.toFixed(2)))
+    setIsPositive(percent_change_usd_last_24_hours >= 0)
+    setPrice(priceFormat.format(price_usd.toFixed(2)))
+    setPercentage(Math.abs(percent_change_usd_last_24_hours.toFixed(2)))
+    setSrcPicture(`https://messari.io/asset-images/${id}/128.png?v=2`)
```
Comment (ale1313) [inline components/CurrencyCard/index.tsx]: [CRITICAL] This host should also be stored on a environment variable

**Review by ale1313 — APPROVED** [review]

**Pull request merged**